configs/module.go

```diff
@@ -425,3 +433,27 @@ func (m *Module) mergeFile(file *File) hcl.Diagnostics {
 
 	return diags
 }
+
+// gatherProviderLocalNames is a helper function that returns a map of provider
+// FQNs -> provider local names. This information is useful for user-facing
+// output, which should include both the FQN and LocalName. It must only be
+// populated after the module has been parsed.
+func (m *Module) gatherProviderLocalNames() {
+	providers := make(map[addrs.Provider]string)
+	for k, v := range m.ProviderRequirements {
+		providers[v.Type] = k
+	}
+	m.ProviderLocalNames = providers
+}
+
+// LocalNameForProvider returns the module-specific user-supplied local name for
+// a given provider FQN, or the default local name if none was supplied.
+func (m *Module) LocalNameForProvider(p addrs.Provider) string {
```

Comment:
Is it worth adding test coverage for this new (unused?) method at this time?

Reply:
Yes! I will add this to `TestNewModule_provider_local_name` - thanks!